### PR TITLE
Add CommentSpacingRule (#3233)

### PR DESCRIPTION
**Pull request by noahsark769**
This PR adds the rule requested in #3233, a rule which enforces double-slash and triple-slash style comments to have at least one space after the slashes. This rule is opt in for now, but is autocorrectable!

I tried to make this an `ASTRule`, but it turns out the `SwiftLintFile`'s `structureDictionary` that is passed to `ASTRule` does not include comment tokens, so I opted to enumerate the comment tokens manually using the `syntaxMap`. Let me know if this isn't the best way to go about this, happy to change it.

I also made a couple of small updates to the messages I encountered during testing to make them more specific/actionable.

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift:4]: This line is over our column limit of 120 characters, which is why almost all of the CI jobs are failing.

File: Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift:101]: Any reason not to include all comment kinds here?

```suggestion
        let commentTokens = file.syntaxMap.tokens.filter { SyntaxKind.commentKinds.contains($0.kind) }
```

File: Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift:108]: This would read better as a raw string literal:

```suggestion
            return regex(#"^(\/){2,3}[^\s\/]"#).matches(in: commentBody, options: .anchored)
```

File: Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift:117]: Why is this 1 rather than 0? Also ByteCount is integer literal expressible:

```suggestion
                            length: 0
```

File: Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift:112]: Nit: use `map` here instead of guard/else/return?

File: Tests/SwiftLintFrameworkTests/TestHelpers.swift
Comment (jpsim) [inline Tests/SwiftLintFrameworkTests/TestHelpers.swift:353]: 👌

**Review by jpsim — APPROVED** [review]
👍 Looks good, with a few nitpick feedback items. Thanks for the helpful contribution!

File: Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift
Comment (noahsark769) [reply Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift:4]: Oops, thanks for calling this out!

File: Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift
Comment (noahsark769) [reply Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift:101]: Ah great call, will definitely put this in

File: Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift
Comment (noahsark769) [reply Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift:117]: I originally thought it should be `1` since the autocorrection looks at the characters and inserts those characters prepending a space, but it occurs to me that we could probably insert a space just fine with 0 length. Will update!

**Comment by jpsim** [comment]
It’s odd that CI didn’t fully run on your last commit. Can you please rebase to fix the CHANGELOG merge conflict and have CI run again?

**Comment by noahsark769** [comment]
Sure! Sorry, I commited those two changes via the Github web UI and meant to address the other issues. Will do that and rebase now

**Comment by noahsark769** [comment]
Alright, rebased and pushed changes. Hopefully CI passes 🤞

File: CHANGELOG.md
Comment (jpsim) [inline CHANGELOG.md:58]: Nit trailing double-space as documented in CONTRIBUTING.md

```suggestion
* Add `comment_spacing` rule.  
```

**Comment by noahsark769** [comment]
Oops, sorry about that! Thanks for fixing 👍

**Review by jpsim — APPROVED** [review]
I made some changes, and just sent you an invitation to have write access to the project.

Would you mind looking over my changes, making any further edits you'd like, and merging when CI passes?

**Comment by noahsark769** [comment]
@jpsim sure, will do! From the git history, looks like we prefer merge rather than squash and merge, so I'll use merge. It looks like CI is failing again, so will try and figure out what's going on

**Comment by noahsark769** [comment]
Changes look great! My bad, I forgot to fix the 1 -> 0 issue.

**Comment by jpsim** [comment]
> It looks like CI is failing again, so will try and figure out what's going on

Oops, my bad. I'll let you fix this.

**Comment by noahsark769** [comment]
```
/__w/1/s/Tests/SwiftLintFrameworkTests/CommandTests.swift:258: error: CommandTests.testDisableAllOverridesSuperfluousDisableCommand : XCTAssertTrue failed - 
/__w/1/s/Tests/SwiftLintFrameworkTests/CommandTests.swift:263: error: CommandTests.testDisableAllOverridesSuperfluousDisableCommand : XCTAssertTrue failed - 
/__w/1/s/Tests/SwiftLintFrameworkTests/CommandTests.swift:268: error: CommandTests.testDisableAllOverridesSuperfluousDisableCommand : XCTAssertTrue failed - 
```

Looks like a couple tests are failing. No big deal, I'll check it out

**Comment by noahsark769** [comment]
There were a few tests which had swiftlint:disable commands with incorrect comment spacing. I fixed those, so hopefully CI should pass now - will merge if it does. Thanks for your help @jpsim!

**Comment by jpsim** [comment]
Thanks for fixing those tests.

**Pull request merged**
